hr_user_ciam_update/models/ciam_update.py

```diff
@@ -7,99 +7,146 @@
 
 _logger = logging.getLogger(__name__)
 
+CIAM_STATUS_OK = 0
+CIAM_STATUS_OBJECT_NOT_FOUND = -3
+
 
 class CIAMUpdate(models.TransientModel):
-    _name = 'hr.employee.user.ciam'
+    _name = "hr.employee.user.ciam"
 
-    employee_id = fields.Many2one('hr.employee', default=lambda self: self.env.context.get('active_id'))
+    employee_id = fields.Many2one(
+        "hr.employee", default=lambda self: self.env.context.get("active_id")
+    )
 
     def action_update_ciam(self):
+        # validate that we have everything we need
         granted = False
-        for group in ('base.group_system',
-                      'base_user_groups_dafa.group_dafa_org_admin_write',
-                      'base_user_groups_dafa.group_dafa_employees_write'):
+        for group in (
+            "base.group_system",
+            "base_user_groups_dafa.group_dafa_org_admin_write",
+            "base_user_groups_dafa.group_dafa_employees_write",
+        ):
             if self.env.user.has_group(group):
                 granted = True
                 break
         if not granted:
             raise AccessError(_("You are not allowed to sync users to CIAM."))
         if not self.employee_id.user_id:
-            raise Warning(_("The employee has no user information. Please assign user groups."))
-        ciam_id = self.env['ciam.client.config'].sudo().search([], limit=1)
+            raise Warning(
+                _("The employee has no user information. Please assign user groups.")
+            )
+        if not self.employee_id.ssnid:
+            raise Warning(_("The employee has no social security number."))
+        legacy_no = self.env["ir.config_parameter"].sudo().get_param("dafa.legacy_no")
+        if not legacy_no:
+            raise Warning("No dafa.legacy_no in system parameters.")
+        ciam_id = self.env["ciam.client.config"].sudo().search([], limit=1)
```

Comment:
`ciam_id` is kind of confusing name. You use it as a client later in the code not as an id for other search params...